### PR TITLE
Drop Python 2 Support

### LINKED ISSUE #393: When will Zarr drop Python 2 support?
Zarr still supports Python 2. As we all know, more and more of the scientific Python ecosystem is moving to Python 3 only. Looking at [PyPI stats](https://pypistats.org/packages/zarr), it seems this compatibility is rarely used. Is now an okay time to discuss when and how to sunset Python 2 support here?

From my limited experience, maintaining Python 2 support makes development a bit harder (and annoying). Particularly when sorting out various string encoding/decoding issues that are relatively common in Zarr.

### PR DESCRIPTION
**Pull request by jhamman**
Drops support for Python 2. Closes #393.

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [x] Changes documented in docs/release.rst
* [x] Docs build locally (e.g., run ``tox -e docs``)
* [x] AppVeyor and Travis CI passes
* [x] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Thanks for doing this @jhamman! We dropped Python 2 support in tskit in [this PR](https://github.com/tskit-dev/tskit/pull/140), which might be useful in outlining the steps required.  In particular, [pyupgrade](https://github.com/asottile/pyupgrade) seems really useful.

**Comment by jhamman** [comment]
Thanks @jeromekelleher - I've implemented the following steps

1. changed setup.py to include `python_requires`
2. removed Python 2 special cases (e.g. `if PY2: ...`)
3. run `pyupgrade`
4. run `isort -rc --atomic .` (helped with step 5)
5. removed `zarr.compat`, imports to py3 modules

Tests are passing locally for me but we'll wait to see what coveralls/travis reports. I also need to add a note in `docs/release.rst`

**Comment by jhamman** [comment]
@zarr-developers/core-devs - just pinging everyone on this. Luckily, we don't have any merge conflicts yet but it would still be good to get this reviewed and merged before things get messy.

**Comment by jhamman** [comment]
@zarr-developers/core-devs - one more ping here. This needs a review.

File: tox.ini [hunk 1]
```diff
@@ -4,15 +4,14 @@
...
 [tox]
-envlist = py27, py35, py36, py37-npy{115,116,latest}, docs
+envlist = py35, py36, py37-npy{115,116,latest}, docs
 
 [testenv]
 install_command = pip install --no-binary=numcodecs {opts} {packages}
 setenv =
     PYTHONHASHSEED = 42
     # hooks for coverage exclusions based on Python major version
     py35,py36,py37: PY_MAJOR_VERSION = py3
```
Comment (alimanfoo) [inline tox.ini]: Actually this line could go as well, it's only needed for marking code to be excluded from coverage when running tests under py2, which we are not running any more.

Comment (jhamman) [reply tox.ini]: Done. Thanks @alimanfo!

**Comment by alimanfoo** [comment]
Thank you Joe, feels like a weight lifted to see this one go in :-)

**Pull request merged**